source/base/polynomials_hermite.cc

```diff
@@ -0,0 +1,122 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 - 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/base/polynomials_hermite.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Polynomials
+{
+  namespace
+  {
+    std::vector<double>
+    hermite_poly_coeffs(const unsigned int regularity, const unsigned int index)
+    {
+      AssertIndexRange(index, 2 * regularity + 2);
+
+      const unsigned int curr_index = index % (regularity + 1);
+      const bool         side       = (index > regularity);
+
+        // Protection needed here against underflow errors
+      const int comparison_1 = static_cast<int>(regularity + 1 - curr_index);
+      const int comparison_2 = side ? static_cast<int>(curr_index + 1) :
+                                      static_cast<int>(regularity + 2);
+
+      std::vector<double> poly_coeffs(2 * regularity + 2, 0.0);
+
+      if (side) // g polynomials
```

Comment:
This this line would turn into the following:
```suggestion
      if (side == 1) // right side: g polynomials
```